### PR TITLE
Mtu on svs dvs

### PR DESCRIPTION
**Pull request by Boredphilosopher96**

### CODE REVIEW

File: vhpc_toolkit/config_objs.py [hunk 1]
```diff
@@ -1157,7 +1160,7 @@ def __init__(self, datacenter_obj):
         self.datacenter_obj = datacenter_obj
         self.logger = log.my_logger(name=self.__class__.__name__)
 
-    def create_dvs(self, host_vmnics, dvs_name, num_uplinks=4):
+    def create_dvs(self, host_vmnics, dvs_name, num_uplinks=4, max_mtu: int = None):
```
Comment (Michael870) [inline vhpc_toolkit/config_objs.py]: rename max_mtu to Stu

File: vhpc_toolkit/config_objs.py [hunk 2]
```diff
@@ -1167,6 +1170,7 @@ def create_dvs(self, host_vmnics, dvs_name, num_uplinks=4):
                                 physical adapters.
             dvs_name (str): The name of the DVS to be created
             num_uplinks (int): Number of active uplinks
+            max_mtu: Max MTU for DVS
```
Comment (Michael870) [inline vhpc_toolkit/config_objs.py]: same as above

File: vhpc_toolkit/get_args.py [hunk 3]
```diff
@@ -972,6 +973,13 @@ def affinity_array(string):
         help="Name of virtual port group to be created within "
         "this standard virtual switch",
     )
+    svs_parser.add_argument(
+        "--mtu",
+        action="store",
+        type=int,
+        help="Max MTU to be set on the SVS. This argument is optional",
```
Comment (Michael870) [inline vhpc_toolkit/get_args.py]: Remove Max. MTU means maximum transmission unit, so "max MTU" doesn't make sense.

File: vhpc_toolkit/get_args.py [hunk 4]
```diff
@@ -1026,6 +1034,13 @@ def affinity_array(string):
         help="Name of virtual port group to be created "
         "within this distributed virtual switch",
     )
+    dvs_parser.add_argument(
+        "--mtu",
+        action="store",
+        type=int,
+        help="Max MTU to be set on the DVS. This argument is optional",
```
Comment (Michael870) [inline vhpc_toolkit/get_args.py]: same as above

File: vhpc_toolkit/operations.py [hunk 5]
```diff
@@ -1813,7 +1813,9 @@ def _create_dvs(self, dvs_cfg):
         for dvs_host in dvs_hosts:
             host_obj = self.objs.get_host(dvs_host)
             host_vmnics[host_obj] = pnics
-        task = ConfigDatacenter(datacenter_obj).create_dvs(host_vmnics, dvs_name)
+        task = ConfigDatacenter(datacenter_obj).create_dvs(
+            host_vmnics, dvs_name, max_mtu=dvs_cfg.get("mtu")
```
Comment (Michael870) [inline vhpc_toolkit/operations.py]: max_mtu -> mtu

**Review by Michael870 — APPROVED** [review]

File: vhpc_toolkit/config_objs.py (hunk 1 above)
Comment (qoofyk) [reply vhpc_toolkit/config_objs.py]: rename max_mtu to mtu, and also change the comment.

File: vhpc_toolkit/config_objs.py (hunk 2 above)
Comment (qoofyk) [reply vhpc_toolkit/config_objs.py]: rename max_mtu to mtu, and also change the comment.

File: vhpc_toolkit/config_objs.py [hunk 6]
```diff
@@ -1229,6 +1233,8 @@ def create_dvs(self, host_vmnics, dvs_name, num_uplinks=4):
             vim.dvs.VmwareDistributedVirtualSwitch.LacpApiVersion.multipleLag
         )
         dvs_config_spec.numStandalonePorts = num_uplinks
+        if max_mtu:
+            dvs_config_spec.maxMtu = max_mtu
```
Comment (qoofyk) [inline vhpc_toolkit/config_objs.py]: It seems that the API use "maxMtu"?
In networking, MTU means maximum transmission unit.

Comment (Michael870) [reply vhpc_toolkit/config_objs.py]: I believe that's a mistake

**Review by qoofyk — APPROVED** [review]

**Pull request merged**